lib/vanagon/component/dsl.rb

```diff
@@ -186,6 +186,8 @@ def install_service(service_file, default_file = nil, service_name = @component.
           target_mode = '0644'
           default_mode = '0644'
         when "smf"
+          # modify version in smf manifest so service gets restarted after package upgrade
+          @component.install << %{#{@component.platform.sed} -ri 's/(<service.*version=)(".*")/\\1"#{@component.settings[:puppet_runtime_version]}"/' #{service_file}}
```

Comment:
I'm not sure we should rely on the runtime_version as you may change something in pxp-agent but if there are no changes on runtime, the version will not be bumped. Same for changing something in runtime that does not affect pxp-agent. It seems that we would mostly do arbitrarily restarts of the services

Reply:
Each nightly puppet agent build is promoted in PE and tested. Of the pxp-agent will be built with the same runtime version, when it ends up in PE, it will not be restored and that could result in the same failures